Re-add the AI core

Doctor-Derp opened the pull request:
## About the Pull Request

Adds back the AI core with a few minor changes, most notably rewiring the core slightly, and adding a suit storage unit and oxygen tanks in place of the suit rack.

 I've also **added an AI upload console to the cyborg upload room**, and added some AI-related rooms back into the areas file.
 ![AI core](https://user-images.githubusercontent.com/40336142/130539488-c97520dc-387b-4181-a3cc-3130ae5f7a9f.PNG)

## Why It's Good For The Game

I've done this at Cupa#3998's request (albeit, 20 days late). AI is still admin-only whitelisted at the time of me making this! And, I personally think people would enjoy/wouldn't mind having AI come back. 

## Did you test it?

It compiled successfully on my computer, and nothing seem amiss when I observed it in-game. AI spawned in successfully, and power was functional. 

## Changelog

 :cl:
 maptweak: Re-adds the AI core, replaces space suit rack with suit storage unit and oxygen tank crate, adds AI upload console to cyborg upload.
/ :cl:

Doctor-Derp committed:
atmospherics fix+AI my beloved+AI airlock fix(?)

Gives the inner and outer cores an atmospherics system and fixes(hopefully) any leaks. I also removed the un-needed room code.

AI my beloved

Adds back the AI core with a few minor changes, most notably rewiring the core slightly, and adding a suit storage unit and oxygen tanks inplace of the suit rack. I've also added an AI upload console to the borg upload, and added some AI-related rooms back into the areas file.

AI airlock fix(?)

I moved things around so where pipes will not intersect with the vents that are causing the problem. I'm unsure if this ACTUALLY fixes it, but we'll see I suppose.

Ilysen commented:
Gonna flag this as DNM until @Cupax3 can give some input on it, since I think we're wanting to do some things leading up to AI actually being re-added.

Cupax3 commented:
Hi,
First of all, thank you a whole bunch for making this PR, it's definitely something that will get merged/is going to be needed, buuut...
... not yet, as we have something planned we want to do before the AI core gets re-added.
I would recommend keeping this as a `Draft PR` until there comes the time that we will add the core back, which is Soon:tm:.
Cheers.

Doctor-Derp commented:
There! I returned it to a draft. I got a little too excited after setting up github and finally finishing the AI core, but I understand :P

Shadowtail117 reviewed (dismissed):
I understand this is a draft, but I'll point out that the CI is failing on the following:

![image](https://user-images.githubusercontent.com/25853190/130682562-4fcc0c66-77ec-498a-a013-9600773552a7.png)

![image](https://user-images.githubusercontent.com/25853190/130682587-67d49f52-b2be-4923-ba12-53ed5e01d433.png)

![image](https://user-images.githubusercontent.com/25853190/130682601-8e4951d0-1615-4e7f-8ae8-ee288b968891.png)

Doctor-Derp commented:
I added in the extra areas just in case someone wanted to make use of an AI upload in the future, should I have not done that? 
And the areas without scrubbers did not have scrubbers in the old core, but I suppose I could add them, and I'm also not sure what's up with the leaking, but i'll look into it when i can.

Shadowtail117 commented:
> I added in the extra areas just in case someone wanted to make use of an AI upload in the future, should I have not done that?
> And the areas without scrubbers did not have scrubbers in the old core, but I suppose I could add them, and I'm also not sure what's up with the leaking, but i'll look into it when i can.

For the former, it's preferred to make areas only when they're needed; if someone will add an AI upload in the future, they can make an area in that PR for it. Keeping unused areas only bloats the game. For the latter two, I'll let you handle it, no rush.

Doctor-Derp commented:
I separated the airlock and life support systems with a multi-layer adapter and different pipes. I also removed the unnecessary areas too! This *should* fix those errors.
![core](https://user-images.githubusercontent.com/40336142/130693241-a6dffa63-e891-46d5-a1e9-ee27cad2fd82.png)

Doctor-Derp commented:
![image](https://user-images.githubusercontent.com/40336142/136997963-7b4b24b2-783d-488b-a637-49d69d1c6925.png)
Changed the layout of the airlock to potentially fix some issues, I'm not entirely sure what's error-ing, but it has something to do with connections. So I moved the pumps around so the supply and scrubber pipes didn't intersect it.

Cupax3 commented:
> Hi, First of all, thank you a whole bunch for making this PR, it's definitely something that will get merged/is going to be needed, buuut... ... not yet, as we have something planned we want to do before the AI core gets re-added. I would recommend keeping this as a `Draft PR` until there comes the time that we will add the core back, which is Soon™️. Cheers.

Consider this void since the plans didn't work out:tm:. This is good to merge from my perspective, depending on if it passes review by Shadow. Note that the merging does not equate people being able to play AI from the get-go, for people passing over this PR with that expectation.

Cupax3 commented:
Test-merge authorized

Cupax3 reviewed (dismissed):
So since this hasn't runtimed into oblivion I presume and it's also just sitting here, consider it approved.

EgorDinamit reviewed (approved):
Commits have been squashed and PR was previously approved by Cupa.
Let's go.
